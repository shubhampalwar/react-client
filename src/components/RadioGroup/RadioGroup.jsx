/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class RadioGroup extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {
      error, options, value, ...rest
    } = this.props;
    return (
      <>
        <h3>What do you do?</h3>
        {options.map(option => (
          <label key={option.value}>
            <input type="radio" {...rest} name={value} value={option.value} />
            {option.label}
            <br />
          </label>
        ))}
        {error ? <p style={{ color: 'red' }}>{error}</p> : ''}
      </>
    );
  }
}

RadioGroup.propTypes = {
  error: PropTypes.string,
  value: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({ label: PropTypes.string, value: PropTypes.string }),
  ),
  onChange: PropTypes.func.isRequired,
};
RadioGroup.defaultProps = {
  error: '',
  options: [],
};
export default RadioGroup;
