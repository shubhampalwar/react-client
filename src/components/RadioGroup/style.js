export default {
  base: {
    padding: '12px',
    margin: '5px 0px 0px 0px',
    borderRadius: '5px',
    border: '1px solid lightgrey',
    width: '100%',
    boxSizing: 'border-box',
  },
  error: {
    border: '1px solid red',
  },
  errorText: {
    color: 'red',
    padding: '10px',
    boxSizing: 'border-box',
  },
  title: {
    fontSize: '21px',
    fontWeight: 'bold',
  },
};
