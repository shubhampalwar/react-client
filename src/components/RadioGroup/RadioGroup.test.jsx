import { shallow } from 'enzyme';
import React from 'react';
import RadioGroup from './RadioGroup';

const wrapper = shallow(<RadioGroup />);

describe('RadioGroup', () => {
  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
