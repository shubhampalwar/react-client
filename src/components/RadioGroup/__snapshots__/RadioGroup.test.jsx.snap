// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioGroup
    error=""
    options={Array []}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          style={
            Object {
              "fontSize": "21px",
              "fontWeight": "bold",
            }
          }
        >
          
        </div>,
        Array [],
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "color": "red",
              "padding": "10px",
            }
          }
        >
          
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "fontSize": "21px",
            "fontWeight": "bold",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "boxSizing": "border-box",
            "color": "red",
            "padding": "10px",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "fontSize": "21px",
                "fontWeight": "bold",
              }
            }
          >
            
          </div>,
          Array [],
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "color": "red",
                "padding": "10px",
              }
            }
          >
            
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "fontSize": "21px",
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "boxSizing": "border-box",
              "color": "red",
              "padding": "10px",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
